src/Interpreters/InterpreterAlterQuery.cpp

```diff
@@ -152,6 +167,30 @@ BlockIO InterpreterAlterQuery::execute()
 }
 
 
+BlockIO InterpreterAlterQuery::executeToDatabase(const ASTAlterQuery & alter)
+{
+    BlockIO res;
+    getContext()->checkAccess(getRequiredAccess());
+    DatabasePtr database = DatabaseCatalog::instance().getDatabase(alter.database);
+    AlterCommands alter_commands;
+
+    for (const auto & child : alter.command_list->children)
+    {
+        auto * command_ast = child->as<ASTAlterCommand>();
+        if (auto alter_command = AlterCommand::parse(command_ast))
+            alter_commands.emplace_back(std::move(*alter_command));
+        else
+            throw Exception("Wrong parameter type in ALTER DATABASE query", ErrorCodes::LOGICAL_ERROR);
+    }
+
+    if (!alter_commands.empty())
+    {
+        database->checkAlterIsPossible(alter_commands, getContext());
+        alter_commands.apply(database, getContext());
```

Comment:
It's better to leave AlterCommands as lightweight class which operates on AST's or some simple structures like `StorageInMemoryMetadata`. Taking IDatabase as argument is not the best solution.